src/lib/vm/core.rs

```diff
@@ -832,9 +832,36 @@ impl VM {
             }
             Primitive::InstVarNamed => unimplemented!(),
             Primitive::InvokeOnWith => todo!(),
-            Primitive::IsDigits => unimplemented!(),
-            Primitive::IsLetters => unimplemented!(),
-            Primitive::IsWhiteSpace => unimplemented!(),
+            Primitive::IsDigits => {
+                let str_val = stry!(rcv.downcast::<String_>(self));
+                let s = str_val.as_str();
+                if s.len() > 0 && s.chars().all(|x| x.is_ascii_digit()) {
+                    self.stack.push(self.true_);
+                } else {
+                    self.stack.push(self.false_);
+                }
+                SendReturn::Val
+            }
+            Primitive::IsLetters => {
+                let str_val = stry!(rcv.downcast::<String_>(self));
+                let s = str_val.as_str();
+                if s.len() > 0 && s.chars().all(|x| x.is_alphabetic()) {
```

Comment:
Same suggestion as above.
```suggestion
                if !s.is_empty() && s.chars().all(|x| x.is_alphabetic()) {
```